src/OgMembershipInterface.php

```diff
@@ -49,4 +54,186 @@
    */
   const REQUEST_FIELD = 'og_membership_request';
 
+  /**
+   * Gets the membership creation timestamp.
+   *
+   * @return int
+   *   The membership creation timestamp.
+   */
+  public function getCreatedTime();
+
+  /**
+   * Sets the membership creation timestamp.
+   *
+   * @param int $timestamp
+   *   The membership creation timestamp
+   *
+   * @return OgMembershipInterface
+   */
+  public function setCreatedTime($timestamp);
+
+  /**
+   * Sets the membership's owner.
+   *
+   * @param mixed $etid
+   *   The user's ID or object.
+   *
+   * @return OgMembershipInterface
+   */
+  public function setUser($etid);
+
+  /**
+   * Gets the membership's owner.
+   * 
+   * @return User
+   *   The user object.
+   */
+  public function getUser();
+
+  /**
+   * Sets the membership field name.
+   *
+   * A user can have two group reference fields. The field name property helps
+   * us to know to which field the membership belongs.
+   *
+   * @param string $fieldName
+   *   The group reference field name.
+   *
+   * @return OgMembershipInterface
+   */
+  public function setFieldName($fieldName);
+
+  /**
+   * Gets the membership field name.
+   *
+   * @return string
+   *   The group reference field name.
+   */
+  public function getFieldName();
+
+  /**
+   * Sets the group entity type ID.
+   *
+   * @param mixed $gid
+   *   The group entity type ID or name.
+   *
+   * @return OgMembershipInterface
+   */
+  public function setEntityId($gid);
```

Comment:
This is setting the entity ID, not the entity type ID.

I think it is confusing to mention that you can pass the "ID or name", this seems to imply you can pass either the ID or the label, but that does not seem to be the intention.